contracts/Tokens/VAI/VAIController.sol

```diff
@@ -387,6 +394,34 @@ contract VAIController is VAIControllerStorageG2, VAIControllerErrorReporter, Ex
         return uint(Error.NO_ERROR);
     }
 
+    /**
+     * @notice Set the prime token contract address
+     * @param prime_ The new address of the prime token contract
+     */
+    function _setPrimeToken(address prime_) external onlyAdmin {
+        emit NewPrime(prime, prime_);
+        prime = prime_;
+
+    }
+
+    /**
+     * @notice Toggle mint only for prime holder
+     * @param _mintEnabledOnlyForPrimeHolder set to true to allow minting only for prime holders
+     * @return uint 0=success, otherwise a failure (see ErrorReporter.sol for details)
+     */
+    function toggleOnlyPrimeHolderMint(bool _mintEnabledOnlyForPrimeHolder) external returns (uint) {
+        _ensureAllowed("toggleOnlyPrimeHolderMint(bool)");
+
+        emit MintOnlyForPrimeHolder(mintEnabledOnlyForPrimeHolder, _mintEnabledOnlyForPrimeHolder);
+        mintEnabledOnlyForPrimeHolder = _mintEnabledOnlyForPrimeHolder;
+
+        if (mintEnabledOnlyForPrimeHolder && prime == address(0)) {
+            return uint(Error.REJECTION);
+        }
```

Comment:
I think we should move this check before the emission of the event to avoid any change in the state